opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt

```diff
@@ -111,6 +110,9 @@ internal object ObservabilityIndex {
                 }
             }
         }
+        if (!this.mappingsUpdated) {
```

Comment:
`else if` here? since if index doesn't exist it will be newly created, and mapping will be new. also set mappingsUpdated to true after creating index

Reply:
Ah, makes sense. Thank you!